### DISCUSSION TITLE
Sending outgoing email via unauthenticated SMTP on localhost port 25

### DISCUSSION THREAD

**Discussion by chriscroome** [mastodon/mastodon#20880] [Q&A]
I have tried a variety settings in `~/live/.env.production`, commenting and uncommenting and amending these variables:

```ini
SMTP_ENABLE_STARTTLS=never
SMTP_DOMAIN=localhost
SMTP_SERVER=localhost
SMTP_PORT=25
SMTP_TLS=false
SMTP_SSL=false
SMTP_LOGIN=
SMTP_PASSWORD=
SMTP_AUTH_METHOD=none
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_CA_FILE=
SMTP_ENABLE_STARTTLS_AUTO=false
SMTP_DELIVERY_METHOD=smtp
```

But I haven't been able to find a combination that allows plain, unencrypted, unauthenticated emails to be passed to the `localhost` where Exim4 is listening on port 25, `STARTTLS` is always used, for example from the Exim4 logs:

```
2022-11-15 07:52:28.605 [140866] SMTP connection from [127.0.0.1]:53160 I=[127.0.0.1]:25 (TCP/IP connection count = 1)
2022-11-15 07:52:28.610 [186225] TLS error on connection from localhost [127.0.0.1]:53160 I=[127.0.0.1]:25 (cert/key setup: cert=/etc/exim4/exim.crt key=/etc/exim4/exim.key): Error while reading file.
2022-11-15 07:52:28.621 [186225] SMTP connection from localhost [127.0.0.1]:53160 I=[127.0.0.1]:25 lost D=0.014s
2022-11-15 07:52:28.621 [186225] no MAIL in SMTP connection from localhost [127.0.0.1]:53160 I=[127.0.0.1]:25 D=0.014s C=EHLO,STARTTLS
```

None of the issues and pull requests I have found related to this seem to have an example of a set of `SMTP_` env vars that work (for example [this one](https://github.com/mastodon/mastodon/pull/2180), [this one](https://github.com/mastodon/mastodon/issues/7323), [this one](https://github.com/mastodon/mastodon/issues/824) and [this one](https://github.com/mastodon/mastodon/issues/772)).

I know that outgoing email via unauthenticated SMTP on localhost port 25 works as I can send email using [Mutt](http://www.mutt.org/) and the following `~/.muttrc`:

```
set smtp_url=smtp://localhost:25/
set ssl_starttls=no
set ssl_force_tls=no
```

Of course `telnet` can also be used&hellip;

Has anyone got outgoing email via unauthenticated SMTP on localhost port 25 working for Mastodon and if so what settings do you have?

**Comment by madeofstown** [comment]
Have you tried using the local IP 127.0.0.1 instead of localhost for the smtp url?

**Comment by leuc** [comment]
Installer sets these local deliver settings

https://github.com/mastodon/mastodon/blob/main/lib/tasks/mastodon.rake#L302
```ruby
env['SMTP_SERVER'] = 'localhost'
env['SMTP_PORT'] = 25
env['SMTP_AUTH_METHOD'] = 'none'
env['SMTP_OPENSSL_VERIFY_MODE'] = 'none'
env['SMTP_ENABLE_STARTTLS'] = 'auto'
```

https://github.com/mastodon/mastodon/blob/main/lib/tasks/mastodon.rake#L365

ActionMailer is being used, detailed config descriptions here:

https://guides.rubyonrails.org/action_mailer_basics.html#action-mailer-configuration

**Comment by ghost** [comment]
Did you setup your system to allow insecure connection? and have you configured it to allow unencrypted connection om port 25?
Also, your email server is trying to read a key file which isn't there, doesn't this make any issues for you?

`2022-11-15 07:52:28.610 [186225] TLS error on connection from localhost [127.0.0.1]:53160 I=[127.0.0.1]:25 (cert/key setup: cert=/etc/exim4/exim.crt key=/etc/exim4/exim.key): Error while reading file.`
